### PR TITLE
126 reverse proxy setup for ssl

### PR DESCRIPTION
**Pull request by hussojo**
Reverse proxy for production build:
- makefile rule for production 'make prod' that builds production.yaml -> nginx works as reverse proxy, back end runs on gunicorn

No websocket handling yet! Coming next..

https://discordapp.com/channels/@me/1113402334928699413/1249007389592649768

### CODE REVIEW

File: production.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+volumes:
+  database:
+
+services:
+  postgres:
+    image: postgres:14
+    container_name: postgres_c
+    user: postgres
+    # ports:
+    #   - 5432:5432
```
Comment (merituulie) [inline production.yaml:15]: Same here! Will these break the tests?

**Comment by dorianjagusch** [comment]
Do we know whether the websockets nonetheless work or does the reverse proxy interfere with that now?

File: frontend/static/js/constants.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
-const baseURL = 'http://localhost:8080/';
+// const baseURL = 'http://localhost:8080/';
+const baseURL = 'http://localhost/';
```
[the remark this answers is not in the document]
Comment (hussojo) [reply frontend/static/js/constants.js:2]: deleted!

File: production.yaml (hunk 1 above)
Comment (hussojo) [reply production.yaml:15]: deleted!

**Comment by hussojo** [comment]
Cleaned comented code!
An to answer Dorian, this works with our project straight away, but it does not exclude possible sockets used for server side pong and cli player! so you can try and 'make prod', nothing brakes.

File: frontend/static/js/constants.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (dorianjagusch) [reply frontend/static/js/constants.js:1]: Is the removal of the port just for prod or also for the dev build?

[the remark this answers is not in the document]
Comment (hussojo) [reply frontend/static/js/constants.js:1]: Good call! so now there is a comment commented out, to remind that the port 8080 needs to be removed when deploying!

**Review by dorianjagusch — APPROVED** [review]
Good work!

**Review by merituulie — APPROVED** [review]

**Pull request merged**